msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -0,0 +1,99 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Xamarin.iOS.Tasks {
+	[TestFixture]
+	public class LocalizationStringTest : TestBase {
+
+		[Test]
+		public void OperatingSystemTranslations ()
+		{
+			CultureInfo culture1 = CultureInfo.CurrentCulture;
+			string locale = culture1.Name;
+
+			var task = CreateTask<CollectITunesArtwork> ();
+			task.ITunesArtwork = new TaskItem [] { new TaskItem (Assembly.GetExecutingAssembly ().Location) };
+
+			Assert.IsFalse (task.Execute (), "Execute failure");
+			Assert.AreEqual (1, Engine.Logger.ErrorEvents.Count, "ErrorCount");
+			bool isTranslated = TranslationAvailable (locale, Engine.Logger.ErrorEvents [0].Message);
+			Assert.IsTrue (isTranslated, "Your current locale is not supported: " + locale + ".");
+		}
+
+		[Test]
+		public void AllSupportedTranslations ()
+		{
+			string reportString = string.Empty;
+			int count = 0;
+			var cultures = new List<string> () {
+				"cs-CZ",
+				"de-DE",
+				"en-US",
+				"es-ES",
+				"fr-FR",
+				"it-IT",
+				"ja-JP",
+				"ko-KR",
+				"pl-PL",
+				"pt-BR",
+				"ru-RU",
+				"tr-TR",
+				"zh-CN",
+				"zh-TW",
+			};
+
+			foreach (string culture in cultures) {
+				CultureInfo newCulture;
+				try {
+					newCulture = new CultureInfo (culture);
+				}
+				catch (System.Globalization.CultureNotFoundException) {
+					reportString += culture + ": is not a valid culture. ";
+					continue;
+				}
+
+				Thread.CurrentThread.CurrentUICulture = newCulture;
```

Comment:
The test should also restore the original current UI culture, otherwise any subsequent tests could end up failing.